Experimental python bindings (#7735)
NixOS/nix: python/doc/api.md

# Experimental Python Bindings

## callExprString

```python
nix.callExprString(expression: str, arg)
```

> Parse a nix expression, then call it as a nix function.

Reviewer: ```suggestion
Parse a Nix expression from the given string, then call it as a Nix function.
```

This needs some more information, such as whether it writes to the store when evaluating derivations and such. Also, how to pass parameters to the evaluator?

Note that this function is experimental and subject to change based on known issues and feedback.

**Parameters:**,

>     `expression` (str): The string containing a nix expression.

Reviewer: ```suggestion
    `expression` (str): The string containing a Nix expression.
```

    `arg`: the argument to pass to the function

**Returns:**

>     `result`: the result of the function invocation, converted to python datatypes.

Reviewer: ```suggestion
    `result`: the result of the function invocation, converted to Python data types.
```

This needs some more information on the data type mapping between Nix and Python.

> # Experimental Python Bindings API

Reviewer 2: ```suggestion
# Experimental Python Bindings
```

A binding is already a specific kind of interface.

> This is the API!

Reviewer 2: TBD.
It'd be good to mention that the function is likely to be replaced.
